receiver/googlecloudmonitoringreceiver/receiver.go

```diff
@@ -5,51 +5,316 @@ package googlecloudmonitoringreceiver // import "github.com/open-telemetry/opent
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+	"time"
 
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+	"cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.uber.org/zap"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	"google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudmonitoringreceiver/internal"
 )
 
 type monitoringReceiver struct {
-	config *Config
-	logger *zap.Logger
-	cancel context.CancelFunc
+	config            *Config
+	logger            *zap.Logger
+	client            *monitoring.MetricClient
+	metricsBuilder    *internal.MetricsBuilder
+	mutex             sync.Mutex
+	metricDescriptors map[string]*metric.MetricDescriptor
 }
 
 func newGoogleCloudMonitoringReceiver(cfg *Config, logger *zap.Logger) *monitoringReceiver {
 	return &monitoringReceiver{
-		config: cfg,
-		logger: logger,
+		config:         cfg,
+		logger:         logger,
+		metricsBuilder: internal.NewMetricsBuilder(logger),
 	}
 }
 
-func (m *monitoringReceiver) Scrape(ctx context.Context) (pmetric.Metrics, error) {
-	// Dummy use to fix lint errors
-	ctx.Deadline()
-	metrics := pmetric.NewMetrics()
-	m.logger.Debug("Scrape metrics ")
+func (mr *monitoringReceiver) Start(ctx context.Context, _ component.Host) error {
+	mr.metricDescriptors = make(map[string]*metric.MetricDescriptor)
 
-	return metrics, nil
-}
+	// Lock to ensure thread-safe access to mr.client
+	mr.mutex.Lock()
+	defer mr.mutex.Unlock()
+
+	// If the client is already initialized, return nil
+	if mr.client != nil {
+		return nil
+	}
+
+	var client *monitoring.MetricClient
+	var err error
+
+	// Use google.FindDefaultCredentials to find the credentials
+	creds, _ := google.FindDefaultCredentials(ctx)
+	// If a valid credentials file path is found, use it
+	if creds != nil && creds.JSON != nil {
+		client, err = monitoring.NewMetricClient(ctx, option.WithCredentials(creds))
+	} else {
+		// Set a default credentials file path for testing
+		os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", "testdata/serviceAccount.json")
```

Comment:
You should do this in test code, rather than in the main logic.

Reply:
You should set this in metadata.yaml to skip tests that require auth setup:

```
tests:
  skip_lifecycle: true
```

Reply:
Then remove this from the receiver.

Reply:
Removed